allennlp/training/optimizers.py

```diff
@@ -112,6 +112,14 @@ def from_params(cls, model_parameters: List, params: Params):  # type: ignore
         else:
             parameter_groups = [param for name, param in model_parameters]
 
+        # Log the number of parameters to optimize
+        num_parameters = 0
+        for parameter_group in parameter_groups:
+            if isinstance(parameter_group, dict):
+                num_parameters += sum(parameter.numel() for parameter in parameter_group["params"])
```

Comment:
Do you need to check for `trainable` somewhere in here?

Reply:
I opted not to, since the logic for that check lives in the trainer. As it's currently provided, all of the parameters in `parameter_groups` get used in the Optimizer --- if some of the input parameters don't `require_grad`, I think the optimizer complains anyway.

Reply:
Ah, ok, great.  This looks good to me, then, but I'd suggest @DeNeutoy also takes a quick look (it's a small PR), just to be sure I'm not missing something.